### PR TITLE
Apply multidex config in kotlin dsl gradle file

### PR DESCRIPTION
**Pull request by GaryQian**
The flake in https://github.com/flutter/flutter/issues/103230 is caused by groovy dynamic dispatch bug. This PR refactors the troublesome logic to Kotlin DSL, which is preferred over Groovy now.

This is an initial step to introduce Kotlin DSL. More refactors may come in the future should we decide to move to it.

Since Kotlin DSL is statically compiled while Groovy is dynamically dispatched, we use withGroovyBuilder and getProperty() to access Groovy metaprogramming.

Since this is a refactor, the existing build tests for multidex and general builds already covers this PR.

### CODE REVIEW

File: packages/flutter_tools/gradle/flutter.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+apply<FlutterPluginKts>()
```
Comment (christopherfujino) [inline packages/flutter_tools/gradle/flutter.gradle.kts:5]: Do we already require of all users the dependencies needed to use Kotlin during builds?

Comment (GaryQian) [reply packages/flutter_tools/gradle/flutter.gradle.kts:5]: I think so but let me double check that.

Comment (GaryQian) [reply packages/flutter_tools/gradle/flutter.gradle.kts:5]: https://docs.gradle.org/5.0/release-notes.html

Kotlin DSL 1.0 is supported in Gradle 5.0 (Nov 2018), and older versions of gradle include pre-1.0 support for Kotlin DSL. This predates Dec 2018 Flutter 1.0 release, so all flutter apps 1.0 and above should support Kotlin DLS buildscripts.

Gradle 3.2 (Nov 2016) contains the initial support for Kotlin DSL (called Gradle Buildscript Kotlin) back then.

Comment (GaryQian) [reply packages/flutter_tools/gradle/flutter.gradle.kts:5]: For reference, Flutter 1.0 shipped with Gradle 4.10.2, which included Kotlin DSL 1.0 RC6 https://docs.gradle.org/4.10.2/release-notes.html

Comment (christopherfujino) [reply packages/flutter_tools/gradle/flutter.gradle.kts:5]: Cool, sounds like Kotlin DSL ships with an embedded compiler: https://docs.gradle.org/current/userguide/migrating_from_groovy_to_kotlin_dsl.html

**Review by christopherfujino — APPROVED** [review]
LGTM let's give it a shot

**Pull request merged**